qiskit_ibm_runtime/visualization/draw_execution_spans.py

```diff
@@ -0,0 +1,115 @@
+# that they have been altered from the originals.
+
+"""Functions to visualize :class:`~.ExecutionSpans` objects."""
+
+from __future__ import annotations
+
+from functools import partial
+from itertools import cycle
+from datetime import datetime, timedelta
+from typing import TYPE_CHECKING
+
+from ..execution_span import ExecutionSpan, ExecutionSpans
+from .utils import plotly_module
+
+if TYPE_CHECKING:
+    from plotly.graph_objects import Figure as PlotlyFigure
+
+
+HOVER_TEMPLATE = "<br>".join(
+    [
+        "<b>ExecutionSpans{id}[{idx}]</b>",
+        "<b>&nbsp;&nbsp;&nbsp;Start:</b> {span.start:%Y-%m-%d %H:%M:%S.%f}",
+        "<b>&nbsp;&nbsp;&nbsp;Stop:</b> {span.stop:%Y-%m-%d %H:%M:%S.%f}",
+        "<b>&nbsp;&nbsp;&nbsp;Size:</b> {span.size}",
+        "<b>&nbsp;&nbsp;&nbsp;Pub Indexes:</b> {idxs}",
+    ]
+)
+
+
+def _get_idxs(span: ExecutionSpan, limit: int = 10) -> str:
+    if len(idxs := span.pub_idxs) <= limit:
+        return str(idxs)
+    else:
+        return f"[{', '.join(map(str, idxs[:limit]))}, ...]"
+
+
+def _get_id(span: ExecutionSpan, multiple: bool) -> str:
+    return f"<{hex(id(span))}>" if multiple else ""
+
+
+def draw_execution_spans(
+    *list_of_spans: ExecutionSpans, common_start: bool = False, normalize_y: bool = False
+) -> "PlotlyFigure":
+    """Draw one or more :class:`~.ExecutionSpans` on a bar plot.
+
+    Args:
+        list_of_spans: One or more :class:`~.ExecutionSpans`.
+        common_start: Whether to shift all collections of spans so that their first span's start is
+            at :math:`t=0`.
+        normalize_y: Whether to display the y-axis units as a percentage of work complete, rather
+            than cummulative shots completed.
+
+    Returns:
+        A plotly figure.
+    """
+    go = plotly_module(".graph_objects")
+    colors = plotly_module(".colors").qualitative.Plotly
+
+    fig = go.Figure()
+    get_id = partial(_get_id, multiple=len(list_of_spans) > 1)
+
+    for spans, color in zip(list_of_spans, cycle(colors)):
+        if not spans:
+            continue
+
+        # sort the spans but remember their original order
+        sorted_spans = sorted(enumerate(spans), key=lambda x: x[1])
+
+        offset = timedelta()
+        if common_start:
+            offset = spans[0].start.replace(tzinfo=None) - datetime(year=1970, month=1, day=1)
+
+        total_size = sum(span.size for span in spans) if normalize_y else 1
+        y_value = 0.0
+        for idx, span in sorted_spans:
+            y_value += span.size / total_size
+            text = HOVER_TEMPLATE.format(span=span, idx=idx, idxs=_get_idxs(span), id=get_id(span))
+            # Create a line representing each span as a Scatter trace
+            fig.add_trace(
+                go.Scatter(
+                    x=[span.start - offset, span.stop - offset],
+                    y=[y_value, y_value],
+                    mode="lines",
+                    line={"width": 4, "color": color},
```

Comment:
It may be useful to allow users to specify this `width`. Users may want to change the size of the figure and if they do, a hardcoded width may make the lines too fat or to slim. In that case, changing the width manually can be inconvenient, especially for users that are not familiar with plotly

Reply:
ab9b77f6ef5a